proposals/resolution_merge.json

```diff
@@ -0,0 +1,73 @@
+{
+    "id": "resolution_merge",
+    "summary": "Resolution merging / pan-sharpening",
+    "description": "Resolution merging algorithms try to improve the spatial resolution of lower resolution bands (e.g. Sentinel-2 20M) based on higher resolution bands (e.g. Sentinel-2 10M).",
+    "categories": [
+        "cubes",
+        "optical"
+    ],
+    "experimental": true,
+    "parameters": [
+        {
+            "name": "data",
+            "description": "Data cube containing multiple spectral bands, with different source resolutions.",
```

Comment:
Is this actually possible? It seems like our current data cube model doesn't allow different resolution bands in a single data cube (i.e. resolution is part of the spectral dimension, not the bands)

Reply:
Indeed, my current implementation actually already upsamples 20m to 10m resolution when loading the data. This process then applies a 'smarter' resolution_merge algorithm.
But the statement that the source resolutions differ is actually true, it's not literally saying that the bands have different resolutions.
So not sure what to do with this in fact...

Reply:
I think my point is that you can't have different resolution bands in reality in our data cubes and thus this process doesn't work as specified here, right? You'd probably need to load two different data cubes with different resolutions and then resolution merge the bands in those data cubes, I guess...

Reply:
Or the same solution as for multiple UTM zones: using additional dimensions to resolve the issue.
Basically what this process does is resolving the issue of loading bands with different resolutions. Backends already do this in fact, but usually the method is basic upsampling. What we do here is simply allowing another method to be specified.

A solution with a merge feels again like pushing more complexity to both users and backends.

Reply:
> Or the same solution as for multiple UTM zones: using additional dimensions to resolve the issue.

Yes, that could be an alternative, I guess...

> Basically what this process does is resolving the issue of loading bands with different resolutions. Backends already do this in fact, but usually the method is basic upsampling. What we do here is simply allowing another method to be specified.

The issue is that it needs some intelligent logic in the process graph parsing, but those side effects are not clear and make the process not very universal. So then the result of the process depends on the underlying "smartness" of the back-end... it would be better if users could actually understand what happens on a step-by-step basis.